File: TypeScript/3.无重复字符的最长字串.ts
function lengthOfLongestSubstring(s: string): number {
    let ans = 0
    const st = new Set<string>()
    let left = 0, right = 0
    while (right < s.length) {
        if (!st.has(s[right])) {
            st.add(s[right])
            ans = Math.max(ans, right - left + 1)
            right++
        } else {
            st.delete(s[left])
            left++
        }
    }
    return ans
};

File: Golang/3.无重复字符的最长字串.go
package leetcode

func lengthOfLongestSubstring(s string) int {
	ans := 0
	m := make(map[byte]bool)
	left, right := 0, 0
	for right < len(s) {
		if _, ok := m[s[right]]; !ok {
			m[s[right]] = true
			ans = max(ans, right-left+1)
			right++
		} else {
			delete(m, s[left])
			left++
		}
	}
	return ans
}

func max(x, y int) int {
	if x < y {
		return y
	}
	return x
}
